examples/python/Earthfile

```diff
@@ -1,12 +1,67 @@
-VERSION --global-cache 0.7
+# Common Python UDCs and Builders.
+VERSION --global-cache --use-function-keyword 0.7
 
-builder:
-    FROM ./../../earthly/python+python-base
+# cspell: words libgcc flake8
 
-    COPY --dir ./src .
-    DO ./../../earthly/python+BUILDER
+python-base:
+    FROM python:3.12-alpine3.19
 
-test:
-    FROM +builder
+    # Install extra packages we will need to support plugins.
+    RUN apk add --no-cache \
+        curl \
+        libffi-dev \
+        gcc \
+        musl-dev \
+        libgcc
 
-    RUN poetry run pytest
+
+    # Set up a virtual environment
+    ENV VIRTUAL_ENV=/opt/venv
+    RUN python3 -m venv $VIRTUAL_ENV
+    ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+    # Poetry Installation directory.
+    # Gives poetry and our poetry project a reliable location.
+    WORKDIR /poetry
+    ENV POETRY_HOME=/poetry
+    ENV PATH=$POETRY_HOME/bin:$PATH
+
+    # Stop python code running in the container saving pointless .pyc files.
+    ENV PYTHONDONTWRITEBYTECODE=1
+    
+    # Install Poetry to manage python.
+    RUN curl -sSL https://install.python-poetry.org | python3 -
+    # Prevent warning when running on machines with more than 6 cores.
+    #   "Connection pool is full, discarding connection: pypi.org. Connection pool size: 10"
+    RUN poetry config installer.max-workers 10
+    
+    # Install flake8 globally using Poetry
+    RUN pip3 install flake8
```

Comment:
+builder should already have the linter installed, so no need to install it again.
We should always use +builder and not redefine the build environment because its not maintainable.

Reply:
Not sure I've understood correctly, but I've tried addressing the duplicate installation in commit c90df2f16b81271f0563faeccfa1d25161ca46af